### PR TITLE
Converge sumtype (offset | repcode) numeric representation towards offBase

### PR DESCRIPTION
**Pull request by Cyan4973**
This completes #2962 , by ensuring that the numeric representation of the sumtype `(offset | repcode)` used at the interface of `ZSTD_storeSeq()` (and elsewhere) is the same as `offBase`, as stored in the `seqStore`.
This skips one transient numeric representation, hopefully simplifying the pipeline.

It's a good opportunity to rename all these transformations to better reflect their position in the pipeline, aka : 
`offset | repcode` => `offBase` => `offCode` + `offBits`
following the model previously established for `matchLength`.

This PR does not impact the compressed output.
The first commit left a subtle minor difference in `btlazy2` parser, resulting in rare minor differences in compressed output. This was later fixed in the last commit, so now there is no difference, even for `btlazy2`.

Also :
fixed `PLATFORM_POSIX_VERSION` within `zstdseek_decompress.c`.
This macro was used there without being defined.
This triggered warning / errors on my workstation during tests.

### CODE REVIEW

**Comment by Cyan4973** [comment]
ping
this should be ready for merge, now that `v1.5.2` is published

**Comment by Cyan4973** [comment]
ping 2

File: contrib/seekable_format/zstdseek_decompress.c [hunk 1]
```diff
@@ -23,13 +23,64 @@
 #  endif
 #endif
 
+/* ************************************************************
```
Comment (terrelln) [inline contrib/seekable_format/zstdseek_decompress.c:26]: Is this change intended? It isn't mentioned in the summary.

**Comment by terrelln** [comment]
Can you please also mention what the first commit, "fixed minor compression difference in btlazy2" is fixing?

It isn't clear what the problem is. And I didn't realize that this PR  had a functional change until I looked at that commit title, I assumed it was all refactoring.

File: contrib/seekable_format/zstdseek_decompress.c (hunk 1 above)
Comment (Cyan4973) [reply contrib/seekable_format/zstdseek_decompress.c:26]: Yes, 
`PLATFORM_POSIX_VERSION` is used a bit below
but it's not defined anywhere,
so it's silently `== 0` by default, always,
or with the right set of compiler flags, it generates a warning, or an error.

Hence, I define it here, using the same definition as in `platform.h`.

**Comment by Cyan4973** [comment]
> Can you please also mention what the first commit, "fixed minor compression difference in btlazy2 " is fixing?
> 
> It isn't clear what the problem is. And I didn't realize that this PR had a functional change until I looked at that commit title, I assumed it was all refactoring.

The PR, as a whole, is "just" a refactoring.
But the first commit introduced a very subtle compression difference in `btlazy2` mode, at line 330.
It's an off-by-one difference in the way cost of offsets are evaluated.
It barely ever triggers, but with billions of attempts, it did happen to change the selection outcome from time to time.
The last commit fixes that, so that the PR as a whole does generate exactly the same compressed outcome as before, including for `btlazy2`.

**Comment by terrelln** [comment]
Can you add that to the PR summary please?

**Comment by Cyan4973** [comment]
> Can you add that to the PR summary please?

done

File: contrib/seekable_format/zstdseek_decompress.c (hunk 1 above)
Comment (terrelln) [reply contrib/seekable_format/zstdseek_decompress.c:26]: Weird... I've been refreshing this page, but GitHub didn't show me this comment until now.

Can you also mention this in the PR summary please? Then the PR LGTM.

Comment (Cyan4973) [reply contrib/seekable_format/zstdseek_decompress.c:26]: done

**Review by terrelln — APPROVED** [review]
